compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -5546,7 +5590,7 @@ private STNode parseObjectMember(SyntaxKind nextTokenKind, STNode metadata) {
                 break;
             case PUBLIC_KEYWORD:
             case PRIVATE_KEYWORD:
-                STNode visibilityQualifier = parseObjectMemberVisibility();
```

Comment:
no we don't. if we reach there all we have to do is just consume that token. Since we reach there we know for sure the token we want is the one we have.

Reply:
What if we reach here by inserting a token as the recovery solution? Then consuming the token will pick the next token from the stream, not the inserted one..

Reply:
yes that is correct, but the method currently doesn't solve this issue. e.g if we get an asterisk token then we still just consume it.

shall I make the method take in next token instead of nextTokenKind, that way we can still use the recovered node.

Reply:
also there is the point that the visibility qualifiers are optional hence recovery won't be inserting them.